CONTRIBUTING.md

```diff
@@ -0,0 +1,25 @@
+# Contributing
+
+We track bugs and issues on individual repos under the
+[cryogen-project](https://github.com/cryogen-project) project.
+There is also an active community on the `#cryogen` channel on the
+[clojurians Slack](https://clojurians.slack.com).
+
+## Bug reports and Issues
+
+If you find a bug or if you want to suggest an enhancement, please
+open an Issue on Github under the appropriate repo.
+
+## Pull Requests
+
+If you want to fix a bug or address an issue:
+
+1. Please fork the repo to your individual account on Github.
+2. Clone that repo to your own machine.
```

Comment:
I'd drop this one. You can do changes directly on GitHub, f.ex. to a Readme, without ever having to clone to a machine.

Reply:
I feel that all changes to the primary repo should be via PRs, and not via direct changes to files on Github.
This keeps all changes grouped properly (if changes are made to multiple files).

I'd prefer that the PRs come from the the individual's repo, not the main repo.
That way, the author is responsible for maintaining or removing the branches.
Otherwise the main repo will be cluttered with branches.

There might be exceptions in case of emergencies, but for the most part, I'd stick with the above.

Moreover, it seems that both of these conventions are what has been followed for this project in practice - once there were more than just one contributor.

Reply:
1. Sorry, I have not explained myself properly. Even if you edit a file on GH, underneath GH will make a branch - and a fork if necessary - and submit a PR. So the process is the same, only the code never touches your machine.
2. "repo will be cluttered with branches" - that is a good argument indeed. Fortunately there is an automated solution - we can configure GH to automatically delete PR branches after a merge. 
3. To be fair, I should present my argument against mandatory forking: it adds extra steps - I have to remember to pull from upstream, not master, and reset to upstream - and from experience I know I mess/miss those up sometimes. Do you have any other arguments for mandatory forks?

Reply:
> Sorry, I have not explained myself properly. Even if you edit a file on GH, underneath GH will make a branch - and a fork if necessary - and submit a PR. So the process is the same, only the code never touches your machine.

I was not aware of this.  It would indeed be good if the code does not touch your machine.  Is this true only for people who have write access to the repo?

From [this documentation](https://docs.github.com/en/free-pro-team@latest/github/managing-files-in-a-repository/editing-files-in-your-repository) I see that Github will make a commit on `master` or create a specific branch on the repo itself.

>  we can configure GH to automatically delete PR branches after a merge.

That works for me.

Perhaps we can recommend (but not enforce) making a fork.  And perhaps we can mention that if making a direct change that a branch and PR is made.  Would that work?

Mostly, I am trying to continue the conventions that are already in place.  Just trying to document them and make them explicit.

Reply:
> I was not aware of this. It would indeed be good if the code does not touch your machine. Is this true only for people who have write access to the repo?
No, you can go to any OSS repo on GH and "edit" a file, which will create a fork and send PR.

> Perhaps we can recommend (but not enforce) making a fork. And perhaps we can mention that if making a direct change that a branch and PR is made. 
Yes!